File: exercises/day16-Santa-Scheduling/typescript/test/santaScheduling.test.ts

describe('Santa Scheduling System Documentation', () => {
  
  describe('TICKET-101: Why does Hawaii get December 25th but New York gets December 24th?', () => {
    
    test('Investigation: What does the code do?', () => {
      // - Then write tests for Hawaii (-10), New York (-5), and others
      
      console.log('Look at santaScheduling.ts - how can we make this testable?');
    });
  });
  
  describe('TICKET-102: Why does London get 8 PM but New York gets 11 PM?', () => {
    
    test('Investigation: Different times for different hemispheres?', () => {
      // TODO: Test London (UTC+0) vs New York (UTC-5)
      // TODO: What causes the 3-hour difference?
    });
  });
  
  describe('TICKET-103: What happens at exactly UTC-5 and UTC+0?', () => {
    
    test('Investigation: Boundary behavior', () => {
      // TODO: Test UTC-5 exactly
      // TODO: Test UTC+0 exactly
      // TODO: Test values just before and after these boundaries
    });
  });
  
  describe('TICKET-104: How does the system handle half-hour timezones?', () => {
    
    test('Investigation: Mumbai and Newfoundland', () => {
      // TODO: Test Mumbai (UTC+5.5)
      // TODO: Test Newfoundland (UTC-3.5)
      // TODO: This was the 2023 incident - what went wrong?
    });
  });
  
  describe('TICKET-105: What\'s the complete rule set?', () => {
    
    test('Investigation: Complete documentation', () => {
      // TODO: Test a range of timezones from -12 to +14
      // TODO: How many distinct behaviors are there?
      // TODO: Can you describe all the rules?
    });
  });
});

File: exercises/day16-Santa-Scheduling/java/src/test/java/SantaSchedulingTest.java
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Nested;
import org.junit.jupiter.api.Test;

@DisplayName("Santa Scheduling System Documentation")
class SantaSchedulingTest {

    /*
     * TODO: Before you can write meaningful tests, you'll need to extract
     * the scheduling logic from Program.cs into a testable method.
     */
    @Nested
    @DisplayName("TICKET-101: Why does Hawaii get December 25th but New York gets December 24th?")
    class Ticket101 {
        
        @Test
        @DisplayName("Investigation: What does the code do?")
        void investigatePattern() {

        }
    }
    
    @Nested
    @DisplayName("TICKET-102: Why does London get 8 PM but New York gets 11 PM?")
    class Ticket102 {
        
        @Test
        @DisplayName("Investigation: Different times for different hemispheres?")
        void investigateTimeDifference() {
            // TODO: Test London (UTC+0) vs New York (UTC-5)
            // TODO: What causes the 3-hour difference?
        }
    }
    
    @Nested
    @DisplayName("TICKET-103: What happens at exactly UTC-5 and UTC+0?")
    class Ticket103 {
        
        @Test
        @DisplayName("Investigation: Boundary behavior")
        void investigateBoundaries() {
            // TODO: Test UTC-5 exactly
            // TODO: Test UTC+0 exactly
            // TODO: Test values just before and after these boundaries
        }
    }
    
    @Nested
    @DisplayName("TICKET-104: How does the system handle half-hour timezones?")
    class Ticket104 {
        
        @Test
        @DisplayName("Investigation: Mumbai and Newfoundland")
        void investigateHalfHourTimezones() {
            // TODO: Test Mumbai (UTC+5.5)
            // TODO: Test Newfoundland (UTC-3.5)
            // TODO: This was the 2023 incident - what went wrong?
        }
    }
    
    @Nested
    @DisplayName("TICKET-105: What's the complete rule set?")
    class Ticket105 {
        
        @Test
        @DisplayName("Investigation: Complete documentation")
        void documentCompleteRules() {
            // TODO: Test a range of timezones from -12 to +14
            // TODO: How many distinct behaviors are there?
            // TODO: Can you describe all the rules?
        }
    }
}
